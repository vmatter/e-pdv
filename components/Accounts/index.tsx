import { useState } from 'react';
import { Button, TextField, Typography, Card } from '@material-ui/core';
import MenuItem from '@material-ui/core/MenuItem';
import {
  Wrapper,
  HeaderWrapper,
  InputWrapper,
  FormHeader
} from './styles';

const Accounts = () => {
  const [showWarning, setShowWarning] = useState(false);
  const [showNameError, setShowNameError] = useState(false);
  const [showPersonError, setShowPersonError] = useState(false);
  const [showEmailError, setShowEmailError] = useState(false);
  const [showPasswordError, setShowPasswordError] = useState(false);
  const [name, setName] = useState('');
  const [person, setPerson] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const onSubmit = async (e: any) => {
    e?.preventDefault();

    if (name === '' || person === '' || email === '' || password === '') {
      name === '' && setShowNameError(true);
      person === '' && setShowPersonError(true);
      email === '' && setShowEmailError(true);
      password === '' && setShowPasswordError(true);
    }
  };

  return (
    <Wrapper>
      <Card variant="outlined" elevation={2}>
        <HeaderWrapper>
          <FormHeader>
            <Typography variant="h4" paddingBottom={1}>
              Gerenciar contas
            </Typography>
            <Typography component="p">
              Crie usuários do tipo Administrador ou Operador de Caixa
            </Typography>
          </FormHeader>
        </HeaderWrapper>
        
        <InputWrapper autoComplete="off">
          <TextField
            id="input-name"
            variant="outlined"
            type="text"
            name="name"
            label="Nome"
            placeholder="ex: João da Silva"
            fullWidth
            helperText={showNameError && 'Este campo deve ser preenchido.'}
            error={showNameError || showWarning}
            onChange={(e: any) => {
              setShowNameError(false);
              setShowWarning(false);
              setName(e.target.value);
            }}
            inputProps={{
              'data-testid': 'input-name',
            }}
          />
         
          <TextField
            id="select-type-person"
            variant="outlined"
            name="typePerson"
            label="Tipo Pessoa"
            fullWidth
            value={person}
            select
            helperText={showPersonError && 'Este campo deve ser preenchido.'}
            error={showPersonError || showWarning}
            onChange={(e: any) => {
              setShowPersonError(false);
              setShowWarning(false);
              setPerson(e.target.value);
            }}
            inputProps={{
              'data-testid': 'select-type-person',
            }}
          >
            <MenuItem value="">Selecione</MenuItem>
            <MenuItem value={"admin"}>Administrador</MenuItem>
            <MenuItem value={"buyer"}>Operador de Caixa</MenuItem>
          </TextField>        
         
          <TextField
            id="input-email"
            variant="outlined"
            type="text"
            name="email"
            label="E-mail"
            placeholder="ex: exemplo@gmail.com"
            fullWidth
            helperText={showEmailError && 'Este campo deve ser preenchido.'}
            error={showEmailError || showWarning}
            onChange={(e: any) => {
              setShowEmailError(false);
              setShowWarning(false);
              setEmail(e.target.value);
            }}
            inputProps={{
              'data-testid': 'input-email',
            }}
          />
         
          <TextField
            id="input-password"
            variant="outlined"
            type="password"
            name="password"
            label="Senha"
            fullWidth
            helperText={showPasswordError && 'Este campo deve ser preenchido.'}
            error={showPasswordError || showWarning}
            onChange={(e: any) => {
              setShowPasswordError(false);
              setShowWarning(false);
              setPassword(e.target.value);
            }}
            inputProps={{
              'data-testid': 'input-password',
            }}
          />
          
          <Button
            id="button-save"
            data-testid="button-save"
            variant="contained"
            color="primary"
            onClick={onSubmit}
          >
            Salvar
          </Button>
        </InputWrapper>
      </Card>
    </Wrapper>
  );
};

export default Accounts;
