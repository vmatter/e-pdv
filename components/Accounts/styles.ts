import styled from '@emotion/styled';
import Snackbar from '@material-ui/core/Snackbar';

export const Wrapper = styled.div`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  overflow: hidden;
`;

export const HeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const FormHeader = styled.div`
  padding: 40px 25px;
`;

export const InputWrapper = styled.form`
  display: flex;
  flex-direction: column;
  padding: 40px 25px;
  gap: 12px;
`;

export const LogoWrapper = styled.a`
  height: 100%;
  margin: auto 25px;
`;

export const LogoImg = styled.img`
  height: 45px;
`;

export const StyledSnackBar = styled(Snackbar)`
  width: 400px;

  ${({ theme }) => `${theme.breakpoints.down('md')} {
    width: 100%;
   }`}
`;