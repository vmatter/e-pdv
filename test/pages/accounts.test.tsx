import Accounts from '../../components/Accounts';
import { render, fireEvent, waitFor } from '../testUtils';

describe('Accounts', () => {

  it('visible fields', async () => {
    const { getByTestId } = render(<Accounts />);

    expect(
      getByTestId('input-name')
    ).toBeInTheDocument();

    expect(
      getByTestId('select-type-person')
    ).toBeInTheDocument();

    expect(
      getByTestId('input-email')
    ).toBeInTheDocument();

    expect(
      getByTestId('input-password')
    ).toBeInTheDocument();


  });

  it('should show erros for required fields', async () => {
    const { getByTestId, getAllByText } = render(<Accounts />);

    fireEvent.click(getByTestId('button-save'));

    await waitFor(() => {
      expect(
        getAllByText('Este campo deve ser preenchido.')[0]
      ).toBeInTheDocument();
    });
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Accounts />, {});
    expect(asFragment()).toMatchSnapshot();
  });

});
