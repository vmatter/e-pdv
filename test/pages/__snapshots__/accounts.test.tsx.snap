// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-y9ds5f-Wrapper e18dzjix5"
  >
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
    >
      <div
        class="css-5m6oo8-HeaderWrapper e18dzjix4"
      >
        <div
          class="css-sdubf6-FormHeader e18dzjix3"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-ggtcmo-MuiTypography-root"
          >
            Gerenciar contas
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1m0ae9b-MuiTypography-root"
          >
            Crie usuários do tipo Administrador ou Operador de Caixa
          </p>
        </div>
      </div>
      <form
        autocomplete="off"
        class="css-1dpfc4n-InputWrapper e18dzjix2"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-16yrtjo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="input-name"
            id="input-name-label"
          >
            Nome
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1g6hcx9-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-weuz2y-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="input-name"
              id="input-name"
              name="name"
              placeholder="ex: João da Silva"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-2s9zix"
              >
                <span>
                  Nome
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-16yrtjo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="select-type-person"
            id="select-type-person-label"
          >
            Tipo Pessoa
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1g6hcx9-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="select-type-person-label select-type-person"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-weuz2y-MuiInputBase-input-MuiOutlinedInput-input"
              id="select-type-person"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              data-testid="select-type-person"
              name="typePerson"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-2s9zix"
              >
                <span>
                  Tipo Pessoa
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-16yrtjo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="input-email"
            id="input-email-label"
          >
            E-mail
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1g6hcx9-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-weuz2y-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="input-email"
              id="input-email"
              name="email"
              placeholder="ex: exemplo@gmail.com"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-2s9zix"
              >
                <span>
                  E-mail
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-16yrtjo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="input-password"
            id="input-password-label"
          >
            Senha
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1g6hcx9-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-weuz2y-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="input-password"
              id="input-password"
              name="password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-2s9zix"
              >
                <span>
                  Senha
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1iqv7g7-MuiButtonBase-root-MuiButton-root"
          data-testid="button-save"
          id="button-save"
          tabindex="0"
        >
          <span
            class="MuiButton-label css-8xplcm-MuiButton-label"
          >
            Salvar
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
